static/src/javascripts/projects/common/modules/email/run-checks.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+import $ from 'lib/$';
+import { keywordExists } from 'lib/page';
+import config from 'lib/config';
+import { getUserAgent } from 'lib/detect';
+import { session } from 'lib/storage';
+import { logError } from 'lib/robust';
+import userPrefs from 'common/modules/user-prefs';
+import { getUserEmailSignUps } from 'common/modules/identity/api';
+
+import type bonzo from 'bonzo';
+import type { ListConfig } from 'common/modules/email/email-article';
+
+let emailShown;
+let userListSubsChecked = false;
+let userListSubs = [];
+
+const pageHasBlanketBlacklist = (): boolean =>
+    // Prevent the blanket emails from ever showing on certain keywords or sections
+    keywordExists(['US elections 2016', 'Football']) ||
+    config.get('page.section') === 'film' ||
+    config.get('page.seriesId') === 'world/series/guardian-morning-briefing';
+
+const userHasRemoved = (id: string, formType: string): boolean => {
+    const currentListPrefs = userPrefs.get(`email-sign-up-${formType}`);
+    return !!currentListPrefs && currentListPrefs.includes(id);
+};
+
+const userHasSeenThisSession = (): boolean =>
+    !!session.get('email-sign-up-seen');
+
+const buildUserSubscriptions = (response: Object): Array<string> => {
+    if (
+        response &&
+        response.status !== 'error' &&
+        response.result &&
+        response.result.subscriptions
+    ) {
+        userListSubs = response.result.subscriptions.map(sub => sub.listId);
+        userListSubsChecked = true;
+    }
+
+    return userListSubs;
+};
+
+const isParagraph = ($el: bonzo): boolean =>
+    !!$el.nodeName && $el.nodeName === 'P';
+
+const allowedArticleStructure = (): boolean => {
+    const $articleBody = $('.js-article__body');
+
+    if ($articleBody.length) {
+        const allArticleEls = $('> *', $articleBody);
+
+        return [].slice
+            .call(allArticleEls, allArticleEls.length - 2)
```

Comment:
```js
allArticleEls.slice(allArticleEls.length - 2)
```

?

Reply:
If `$` returns a `bonzo` can you call `slice` directly on a bonzo list? 

You could try `[...allArticleEls].slice(allArticleEls.length - 2)`

Reply:
The bad news: neither of those patterns works for `bonzo` 🤖 
The good news `Array.from(...)` does work 🤠